File: 프로그래머스/2/42586. 기능개발/기능개발.py
def solution(progresses, speeds):
    answer = []
    
    # Calculate the remaining days for each task
    days = []
    for i, progress in enumerate(progresses):
        remains =  (100 - progress)
        # Calculate the required days considering the speed
        day = (remains // speeds[i]) if (remains % speeds[i]) == 0 else (remains // speeds[i]) + 1
        days.append(day)
    
    # Count the number of tasks that can be released together
    while days:
        # Take the first element as the benchmark
        top = days.pop(0)
        cnt = 1
        
        # Check subsequent tasks if they can be released together
        while days and top >= days[0]:
            days.pop(0)
            cnt += 1
        
        answer.append(cnt)
            
    return answer

File: 프로그래머스/2/42586. 기능개발/기능개발.cpp
#include <string>
#include <vector>
#include <cmath>
#include <queue>

using namespace std;

vector<int> solution(vector<int> progresses, vector<int> speeds) {
    vector<int> answer;
    
    queue<int> q;
    for(int i=0; i < speeds.size(); i++){
        double rest = 100.0 - static_cast<double>(progresses[i]);
        int finish = static_cast<int>(ceil(rest / static_cast<double>(speeds[i])));
        q.push(finish);
    }
    
    while(!q.empty()){
        int top = q.front(); q.pop();
        int cnt = 1;
        
        while(!q.empty() && top >= q.front()){
            q.pop();
            cnt += 1;
        }
        
        answer.push_back(cnt);
    }
    
    return answer;
}